### ISSUE TITLE
Bug: renaming and moving a directory results in confusing commit subject

### ISSUE THREAD

**Issue by aqw** [psyinfra/onyo#709]
I'm not sure if there's more than one misbehavior here, but at minimum, the commit subject is confusing.

Let's say I move a directory and rename it at the destination, such as:
```
onyo mv groups/dp/Alex\ Waite groups/psyinf/Alex\ Stellfox
```

This results in a commit subject of:
```
mv + ren [2]: Alex Waite,Alex Waite -> Alex Stellfox
```
The `Alex Waite,Alex Waite -> Alex Stellfox` is just weird. I /think/ the first `Alex Waite` is for `mv` (which is meaningless), and the second is for the `ren` (which is different than normal `ren` messages). I would like to see a rendering such as the one outlined in #708. However, I don't know if that's reasonable to expect if we indeed consider this to be two different operations.

In short, should this be considered one or two operations?
1) if one, then this should render like any other rename (which perhaps #708 will improve upon)
2) if two, I don't have a suggestion yet, and we need to some up with a more meaningful message

**Comment by TobiasKadelka** [comment]
I think you meant to provide an example for which the directory changes, too? Because `onyo mv groups/dp/Alex\ Waite groups/dp/Alex\ Stellfox` does not move the directory, it just renames it. For just renaming, it is no move, I only get `ren [1]: groups/dp/Alex\ Waite groups/dp/Alex\ Stellfox`.

`onyo mv groups/dp/Alex\ Waite groups/sbb/Alex\ Stellfox` leads to 
`mv + ren [2]: Alex Waite,Alex Waite -> Alex Stellfox`

For this, I personally do think two operations makes sense. It is only on the terminal/computer level that it is one operation, but in real life this a rather rare situation, and I think it will almost always be caused by two things happening at once. (The only exception might be if we completely would change our whole inventory structure maybe, e.g. rename groups, merge them, change from the current `spaces/` to something different. But I do not think that in every day operation the two things happen often at once)

--------------

For assets, I really do think this should be considered and displayed as two operations in the history.

When I reflect on when this would happen, it seems always like a short form of we received the device, changed/repaired it, and gave it out to a new person, or some other action like that, that would describe multiple actions in real life.

So in summary, I think a rename+move in one call does reflect two actions, and should be displayed as such.
I only think we should make the commit message subject more meaningful, for example even a `groups/dp/Alex\ Waite -> groups/sbb/Alex\ Stellfox` feels more correct an meaningful.

**Comment by aqw** [comment]
> I think you meant to provide an example for which the directory changes, too?

Ahh yes. Sorry about that. I copy-pasted the wrong command. That is fixed now.

> When I reflect on when this would happen, it seems always like a short form of we received the device, changed/repaired it, and gave it out to a new person, or some other action like that, that would describe multiple actions in real life.

But those will all be moves.

A `mv` + `ren` can only happen to a non-asset directory.

The question is: what is the directory name, is it:
1) `groups/dp/Alex Waite`
2) `Alex Waite`

If it's the first one, then changing `groups/dp/Alex Waite` -> ``groups/psyinf/Alex Stellfox` is just a rename.

If it's the second, then it's two actions. Move then rename.

**Comment by TobiasKadelka** [comment]
> Your examples are with Assets.

That is not true, just the examples after the deviding line are (and you are right, this can't happen). But the whole part before is about directories. And the complete "in summary" part describes all move+rename scenarios.

And I think *it is* the second case. Those are always two operations in the real world, and it is way more clear in messages when it is displayed clearly as "two things happened at once".

Your example makes it even more clear for me, because I think about `Alex Waite/` as the directory, because it is the location about which I think when considering where something is. `groups/dp/` is more a pseudo key for me.

Considering #693 (which is about assets, not dirs, I know), there might be an argument for considering both operations (mv+rename) of a directory as the equivalent of setting `onyo.path.parent` (just for a directory instead). But I don't know. When just thinking about this issue and how it connects to real world inventorying, it simply feels more right to move+rename a directory, and consider this as two separate operations, and reflect this in the subject (to resolve this issue, the subject should be changed to something like `groups/dp/Alex\ Waite -> groups/sbb/Alex\ Stellfox`).

**Comment by bpoldrack** [comment]
I think this discussion is a bit derailed.
Onyo tracks dirs. A dir is a meaningful inventory item. Hence, there's no question to me that this is in fact two operations.
But the issue is about the subject line of the commit, not about the operations record.

And the current form comes from: `groups/dp/Alex\ Waite -> groups/psyinf/Alex\ Waite` + `groups/psyinf/Alex\ Waite -> groups/psyinf/Alex\ Stellfox` is too long.
But the shortening does indeed result in something that doesn't read nice at all. It's shortening to `mv: Alex Waite` which would be nice and correct if it wasn't for the additional rename. Because that is the item that was moved.

With #708, I'd say:

`mov + ren [1]: groups/{dp -> psyinf}/{Alex Waite -> Alex Stellfox}`

might be a solution. Implying what I will put there, too: The same shortening pattern suggested for rename is likely useful for a plain move as well.

**Comment by aqw** [comment]
> With #708, I'd say:
> `mv+ren [1]: groups/{dp -> psyinf}/{Alex Waite -> Alex Stellfox}`

This would be marvelous.

**Comment by aqw** [comment]
With PR #705, this matches the desired commit message:
```
mv + ren [2]: groups/{dp -> psyinf}/{Alex Waite -> Alex Stellfox}
```
